deps/spidershim/src/v8context.cc

```diff
@@ -139,6 +138,52 @@ void Context::Exit() {
   GetIsolate()->PopCurrentContext();
 }
 
+void Context::SetEmbedderData(int idx, Local<Value> data)
+{
+	assert(idx >= 0);
+  if (pimpl_->embedderData.length() <= static_cast<unsigned int>(idx) &&
+      !pimpl_->embedderData.resize(idx + 1)) {
+    return;
+  }
+
+  pimpl_->embedderData[idx].set(*GetValue(data));
+}
+
+Local<Value>
+Context::GetEmbedderData(int idx)
+{
+	assert(idx >= 0);
+  if (static_cast<unsigned int>(idx) >= pimpl_->embedderData.length()) {
+    return Local<Value>();
+  }
+
+  return Local<Value>::New(Isolate::GetCurrent(), GetV8Value(pimpl_->embedderData[idx]));
+}
+
+void
+Context::SetAlignedPointerInEmbedderData(int idx, void* data)
+{
+  assert((reinterpret_cast<uintptr_t>(data) & 0x1) == 0);
+  assert(idx >= 0);
+  if (static_cast<unsigned int>(idx) >= pimpl_->embedderData.length() &&
+		  !pimpl_->embedderData.resize(idx + 1)) {
+    return;
+  }
+
+  pimpl_->embedderData[idx].set(JS::PrivateValue(data));
+}
+
+void*
+Context::GetAlignedPointerFromEmbedderData(int idx)
+{
+	assert(idx >= 0);
+  if (static_cast<unsigned int>(idx) >= pimpl_->embedderData.length()) {
+    return nullptr;
+  }
+
+  return pimpl_->embedderData[idx].get().toPrivate();
```

Comment:
On Thu, May 12, 2016 at 03:41:17PM -0700, Ehsan Akhgari wrote:
> In deps/spidershim/src/v8context.cc:
> 
> > +               !pimpl_->embedderData.resize(idx + 1)) {
> > +    return;
> > +  }
> > +
> > +  pimpl_->embedderData[idx].set(JS::PrivateValue(data));
> > +}
> > +
> > +void*
> > +Context::GetAlignedPointerFromEmbedderData(int idx)
> > +{
> > +     assert(idx >= 0);
> > +  if (static_cast<unsigned int>(idx) >= pimpl_->embedderData.length()) {
> > +    return nullptr;
> > +  }
> > +
> > +  return pimpl_->embedderData[idx].get().toPrivate();
> 
> You need to also check isPrivate() above and return null if it fails.

there is no isPrivate() I think waldo said that is because it uses the
same tag as int32_t.  I'm not sure v8 checks that either.  I'm not sure
this is right, but having examined what node uses this for I can say it
doesn't matter for that use case.